aea/manager.py

```diff
@@ -0,0 +1,534 @@
+        """Remove agent alias from project."""
+        self.project.agents.remove(self.name)
+
+
+class AsyncTask:
+    """Async task wrapper for agent."""
+
+    def __init__(self, agent: AEA, loop: asyncio.AbstractEventLoop) -> None:
+        """Init task with agent and loop."""
+        self.run_loop: asyncio.AbstractEventLoop = loop
+        self.caller_loop: asyncio.AbstractEventLoop = loop
+        self._done_future: Optional[asyncio.Future] = None
+        self.task: Optional[asyncio.Task] = None
+        self.agent = agent
+
+    def create_run_loop(self) -> None:
+        """Create run loop."""
+        pass
+
+    def start(self) -> None:
+        """Start task."""
+        self.create_run_loop()
+        self.task = self.run_loop.create_task(self._run_wrapper())
+        self._done_future = asyncio.Future(loop=self.caller_loop)
+
+    def wait(self) -> asyncio.Future:
+        """Return future to wait task completed."""
+        if not self._done_future:
+            raise ValueError("Task not started!")
+        return self._done_future
+
+    def stop(self) -> None:
+        """Stop task."""
+        if not self.run_loop or not self.task:
+            raise ValueError("Task was not started!")
+        self.run_loop.call_soon_threadsafe(self.task.cancel)
+
+    async def _run_wrapper(self) -> None:
+        """Run task internals."""
+        if not self._done_future:
+            raise ValueError("Task was not started! please use start method")
+        try:
+            await self.run()
+            self.caller_loop.call_soon_threadsafe(self._done_future.set_result, None)
+        except asyncio.CancelledError:
+            self.caller_loop.call_soon_threadsafe(self._done_future.set_result, None)
+        except Exception as e:
+            self.caller_loop.call_soon_threadsafe(self._done_future.set_exception, e)
+
+    async def run(self) -> None:
+        """Run task body."""
+        self.agent.runtime.set_loop(self.run_loop)
+        try:
+            self.agent.runtime.start()
+            while True:
+                await asyncio.sleep(1)
+        finally:
+            self.agent.runtime.stop()
+
+    @property
+    def is_running(self) -> bool:
+        """Return is task running."""
+        return not self.wait().done()
+
+
+class ThreadedTask(AsyncTask):
+    """Threaded task."""
+
+    def __init__(self, agent: AEA, loop: asyncio.AbstractEventLoop) -> None:
+        """Init task with agent and loop."""
+        AsyncTask.__init__(self, agent, loop)
+        self._thread: Optional[Thread] = None
+
+    def create_run_loop(self) -> None:
+        """Create run loop."""
+        self.run_loop = asyncio.new_event_loop()
+
+    def start(self) -> None:
+        """Run task in a dedicated thread."""
+        super().start()
+        self._thread = threading.Thread(
+            target=self.run_loop.run_until_complete, args=[self.task], daemon=True
+        )
+        self._thread.start()
+
+
+class Manager:
```

Comment:
`MultiAgentManager` is how we will call it now

Reply:
Docstring needs updating; it is not abstract anymore

Reply:
@solarw

Reply:
updated